ampd/src/broadcaster_v2/msg_queue.rs

```diff
@@ -117,23 +119,32 @@ where
     /// * `Error::EnqueueMsg` - If enqueueing fails
     /// * `Error::GasExceedsGasCap` - If the message requires more gas than allowed
     /// * `Error::ReceiveTxResult` - If the result channel is closed prematurely
-    pub async fn enqueue(
-        &mut self,
-        msg: Any,
-    ) -> Result<impl Future<Output = Result<(String, u64)>> + Send> {
-        let attachment = json!({ "msg": &msg });
+    #[instrument(skip(self))]
+    pub async fn enqueue(&mut self, msg: Any) -> Result<impl Future<Output = TxResult> + Send> {
         let rx = self
-            .enqueue_with_channel(msg)
+            .enqueue_with_channel(msg.clone())
             .await
-            .map_err(|err| err.attach_printable(attachment.clone()))?;
+            .inspect_err(|err| {
+                error!(
+                    err = LoggableError::from(err).as_value(),
+                    msg = ?msg,
+                    "failed to enqueue message"
+                );
+            })?;
 
         Ok(rx
             .map(|result| match result {
                 Ok(Ok(result)) => Ok(result),
                 Ok(Err(err)) => Err(err),
-                Err(err) => Err(err.into_report()),
+                Err(err) => Err(Arc::new(err.into_report())),
             })
-            .map_err(move |err| err.attach_printable(attachment)))
+            .inspect_err(move |err| {
+                error!(
+                    err = LoggableError::from(err.as_ref()).as_value(),
+                    msg = ?msg,
+                    "failed to receive tx result"
+                );
+            }))
```

Comment:
we're still passing the error on here, right? Should we have logging here or move it to the caller?